crates/client/db/src/rocksdb_options.rs

```diff
@@ -1,3 +1,50 @@
+//! # Rocksdb options
+//!
+//! We configure the rocksdb database in a very specific way. We want to guarantee a few things:
+//!
+//! ## Fault tolerance
+//!
+//! The madara node can be stopped at any time, and it should be able to recover when restarting.
+//! In particular, in case of a power/hardware failure, the database should not be corrupted beyond repair. We cannot
+//! run code that would allow us to gracefully shutdown in these cases.
```

Comment:
> We cannot run code that would allow us to gracefully shutdown in these cases.

Is that correct? I feel like there's a "not" missing, or something.

Reply:
unsure if my grammar is correct but it looks correct to me: if there's a power failure, or the node is killed with SIGKILL for example, there is no graceful shutdown - the program just abruptly stops

in WAL mode, rocksdb will look at the WAL on startup and fix the potential corruption (ie. the node stopped during a write)
in case of a graceful shutdown (SIGTERM/SIGINT/...) the rocksdb Arc<DB> gets dropped, and in that destructor the WAL is flushed and there is no need to recover from WAL the next time on startup